### DISCUSSION TITLE
Adding New interpolator to pybamm.Interpolant class.

### DISCUSSION THREAD

**Discussion by MartinFuentetaja** [pybamm-team/PyBaMM#4589] [Q&A]
Hello everyone! I am trying to use Akima1D interpolator to interpolate the OCPs of gr/LFP cell. I have tried to use pchip instead of cubic spline (due to the flatness of the OCP of LFP). However, the following error arises:

NotImplementedError: The interpolator 'pchip' is not supported by CasAdi. Use 'linear' or 'cubic' instead. Alternatively, set 'model.convert_to_format = 'python'' and use a non-CasADi solver. 

Therefore, I have also tried to implement Akima1D interpolator from scratch. However, I did not success. 

I have to mention that it works with linear interpolator, however I would prefer to use another one.

**Comment by martinjrobins** [comment]
This is possible, the main piece of work would be to implement the pchip interpolant in the file `pybamm/expression_tree/operations/convert_to_casadi.py`. There is an issue on the casadi repo that gives an example of how to do this: https://github.com/casadi/casadi/issues/2489

**Comment by vipulramknovo** [comment]
Hello @MartinFuentetaja could you please share the code block you used as a interpolation function also how you updated in the parameter values. 
I am have some trouble using the interpolation function as as a function of stoichiometry

**Reply by MartinFuentetaja** [reply]
Hello @vipulramknovo, I'm sorry, but I can't share the code block. However, if it helps, I used the [Okane2022.py](https://github.com/pybamm-team/PyBaMM/blob/5e6678bb691bb383e8ae5746eb5ca7f251a73d4e/src/pybamm/input/parameters/lithium_ion/OKane2022.py#L68) script as a reference to develop my own methods for an LFP/Graphite cell. You can check it out to see how it can be implemented.

I hope this is helpful!
